diff --git i/ipkg.py w/ipkg.py
index a4f440e..9aadf45 100644
--- i/ipkg.py
+++ w/ipkg.py
@@ -36,12 +36,18 @@ import tempfile
 import os
 import sys
 import glob
-import md5
 import re
 import string
 import commands
 from stat import ST_SIZE
 
+try:
+    from hashlib import md5
+except ImportError:
+    import md5
+    md5 = md5.new
+
+
 class Package:
     """A class for creating objects to manipulate (e.g. create) ipkg
        packages."""
@@ -77,7 +83,7 @@ class Package:
 
             # compute the MD5.
             f = open(fn, "r")
-            sum = md5.new()
+            sum = md5()
             while 1:
                 data = f.read(1024)
                 if not data: break
